src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -65,6 +74,23 @@ public FObject put_(X x, FObject obj) {
       toPut.setSpid(user.getSpid());
     }
 
+    User result = (User) super.find_(x, toPut.getId());
+
+    if (
+      result != null &&
+      ! SafetyUtil.equals(result.getGroup(), toPut.getGroup())
+    ) {
+      boolean hasGroupUpdatePermission = auth.check(x, "group.update." + result.getGroup());
+      if ( updatingSelf ) {
+        throw new AuthorizationException("You cannot change your own group.");
+      } else if ( ! hasUserEditPermission ) {
+        throw new AuthorizationException("You do not have permission to change this user's group.");
+      } else if ( ! hasGroupUpdatePermission ) {
+        throw new AuthorizationException("You do not have permission to change a user's group to '" + toPut.getGroup() + "'.");
+      }
```

Comment:
I'd rather see this logic in a separate decorator as to not bloat the AuthenticatedUserDAO too much

Reply:
But this logic is related to authorization, so I think it would be confusing to have that responsibility split betwen two AuthenticatedUserDAOs.